server/queries/BattleQueries.cpp

```diff
@@ -104,6 +98,12 @@ void CBattleDialogQuery::onRemoval(PlayerColor color)
 	}
 	else
 	{
+		auto hero = bi->getSideHero(BattleSide::ATTACKER);
+		auto visitingObj = bi->getDefendedTown() ? bi->getDefendedTown() : gh->getVisitingObject(hero);
+		
 		gh->battles->endBattleConfirm(bi->getBattleID());
+		
+		if(visitingObj)
+			visitingObj->battleFinished(hero, *result);
```

Comment:
How will the game call `battleFinished` in case of AI-initiated battles? 

From what I see, `CBattleDialogQuery` is only created for AI-vs-human battles. And not created in AI-vs-AI or human-vs-human battles

Reply:
Yes, you're right. Changed it... new way only for AI vs human.